warehouse/warehouse.go

```diff
@@ -655,6 +676,60 @@ func (wh *HandleT) SyncLoadFilesToWarehouse(job *ProcessStagingFilesJobT) (err e
 	return
 }
 
+func (wh *HandleT) recordDeliveryStatus(uploadID int64) {
+	var (
+		sourceID      string
+		destinationID string
+		status        string
+		errorCode     string
+		errorResp     string
+		updatedAt     time.Time
+		tableName     string
+		tableStatus   string
+	)
+	successfulTableUploads := make([]string, 0)
+	failedTableUploads := make([]string, 0)
+
+	row := wh.dbHandle.QueryRow(fmt.Sprintf(`select source_id, destination_id, status, error, updated_at from %s where id=%d`, warehouseUploadsTable, uploadID))
+	row.Scan(&sourceID, &destinationID, &status, &errorResp, &updatedAt)
+	rows, _ := wh.dbHandle.Query(fmt.Sprintf(`select table_name, status from %s where wh_upload_id=%d`, warehouseTableUploadsTable, uploadID))
+	for rows.Next() {
+		rows.Scan(&tableName, &tableStatus)
+		if tableStatus == warehouseutils.ExportedDataState {
+			successfulTableUploads = append(successfulTableUploads, tableName)
+		} else {
+			failedTableUploads = append(failedTableUploads, tableName)
+		}
+	}
+	var errorRespB []byte
+	if errorResp == "{}" {
+		errorCode = "200"
+		errorRespB, _ = json.Marshal(ErrorResponseT{Error: errorResp})
+	} else {
+		errorCode = "400"
+		errorRespB = []byte(`{"success":"OK"}`)
```

Comment:
shouldn't this `errorRespB, _ = json.Marshal(ErrorResponseT{Error: errorResp})` be here?

Reply:
also, we do not reset the error if it succeeds in later attempts. In those cases we should not send the error I guess